examples/misc/programs/misc/src/lib.rs

```diff
@@ -80,6 +80,20 @@ pub mod misc {
         Ok(())
     }
 
+    pub fn test_simulate_associated_account(
+        ctx: Context<TestSimulateAssociatedAccount>,
+        data: u32,
+    ) -> ProgramResult {
+        let associated_account = *ctx.accounts.my_account.to_account_info().key;
```

Comment:
Was not sure how to test creation of the associated account. Tried to check in the client whether the account had been created like so:

```
const account = await program.account.testData.associated(
  program.provider.wallet.publicKey,
);
```
But it did not work. My guess is that `program.simulate()` clashes with that check ?
As a workaround, I am passing the pubkey back in an event and checking that it matches.